spring-boot/admin/src/main/java/eu/coatrack/admin/controllers/GatewayApiController.java

```diff
@@ -51,4 +69,37 @@ public ApiKey findApiKeyEntityByApiKeyValue(@RequestParam("keyValue") String api
     public ServiceApi findServiceByApiKeyValue(@RequestParam("apiKeyValue") String apiKeyValue) {
         return serviceApiRepository.findByApiKeyValue(apiKeyValue);
     }
+
+    //Due to legacy code reasons the gateway API key and the gateway id are exactly the same.
+    @GetMapping("/api/gateways/api-keys")
+    public ResponseEntity<?> findApiKeyListByGatewayApiKey(@RequestParam("gateway-api-key") String gatewayIdAndApiKey) {
+        log.debug("The gateway with the ID {} requests its latest API key list.", gatewayIdAndApiKey);
+        try {
+            Optional<Proxy> proxy = Optional.of(proxyRepository.findById(gatewayIdAndApiKey));
+            return createResponseEntityIfProxyWasFound(proxy, gatewayIdAndApiKey);
+        } catch (Exception e) {
+            log.info("The creation of the API key list for the gateway {} failed.", gatewayIdAndApiKey, e);
+            return new ResponseEntity<>(e, HttpStatus.NOT_FOUND);
+        }
+    }
+
+    private ResponseEntity<?> createResponseEntityIfProxyWasFound(Optional<Proxy> proxy, String gatewayIdAndApiKey)
```

Comment:
This looks over-engineered. If you would put this code into the calling method, it would be easier to understand plus less lines of code in total.

Reply:
I agree. Done.